File: C++17/백준/Gold/14786. Ax＋Bsin（x）＝C ②/Ax＋Bsin（x）＝C ②.cc
#include <cstdio>
#include <cmath>
using namespace std;

int main() {
    double A, B, C;
    scanf("%lf %lf %lf", &A, &B, &C);

    double s = 0, e = 100000;
    while (e - s > 1e-9) {
        double mid = (s + e) / 2;
        double val = A * mid + B * sin(mid);
        if (val * val > C * C)
            e = mid;
        else if (val * val < C * C)
            s = mid;
        else break;
    }

    printf("%.15lf\n", e);

    return 0;
}

File: PyPy3/백준/Gold/14786. Ax＋Bsin（x）＝C ②/Ax＋Bsin（x）＝C ②.py
import sys
import math
input = sys.stdin.readline

A, B, C = map(int, input().split())

s, e = 0, 100_000

while e - s > 1e-9:
    mid = (s + e) / 2
    
    if (A * mid + B * math.sin(mid)) ** 2 > C ** 2:
        e = mid
    elif (A * mid + B * math.sin(mid)) ** 2 < C ** 2:
        s = mid
    else:
        break
    
print(e)
